group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -8686,6 +8695,88 @@ public void testListGroups() {
         assertEquals(expectAllGroupMap, actualAllGroupMap);
     }
 
+    @Test
+    public void testDescribeGroupStableForDynamicMember() throws Exception {
+        GroupMetadataManagerTestContext context = new GroupMetadataManagerTestContext.Builder()
+            .build();
+        JoinGroupResponseData responseData = context.joinGenericGroupAsDynamicMemberAndCompleteRebalance("group-id");
+
+        List<DescribeGroupsResponseData.DescribedGroup> describedGroups =
+            context.describeGroups(Collections.singletonList("group-id"));
+
+        assertEquals(responseData.protocolType(), describedGroups.get(0).protocolType());
+        assertEquals(responseData.protocolName(), describedGroups.get(0).protocolData());
+        assertEquals(responseData.members().stream().map(member -> member.memberId()).collect(Collectors.toList()),
+            describedGroups.get(0).members().stream().map(member -> member.memberId()).collect(Collectors.toList())
+        );
+    }
+
+    @Test
+    public void testDescribeGroupStableForStaticMember() throws Exception {
+        GroupMetadataManagerTestContext context = new GroupMetadataManagerTestContext.Builder()
+            .build();
+        context.staticMembersJoinAndRebalance(
+            "group-id",
+            "leader-instance-id",
+            "follower-instance-id"
+        );
+        GenericGroup group = context.groupMetadataManager.getOrMaybeCreateGenericGroup("group-id", false);
```

Comment:
nit: Let's move this one to right before the assertions as it is only used there.